### ISSUE TITLE
Having 'watchOS' support in podspec with 'test_spec' causes spec linting failure

### ISSUE THREAD

**Issue by maximkhatskevich** [CocoaPods/CocoaPods#8283]
* [X] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I have a lib project (cross-platform Cocoa framework).

I use Bundler, here is my `Gemfile`:

```ruby
source "https://rubygems.org"

gem 'fastlane'
gem 'cocoapods', '~> 1.6.0.beta.2'
gem 'cocoapods-generate'
```

Here is my `podspec` file:

```ruby
Pod::Spec.new do |s|

    s.name          = 'XCEPipeline'
    s.summary       = 'Custom pipeline operators for easy chaining in Swift.'
    s.version       = '1.1.2'
    s.homepage      = 'https://XCEssentials.github.io/Pipeline'

    s.source        = { :git => 'https://github.com/XCEssentials/Pipeline.git', :tag => s.version }

    s.requires_arc  = true

    s.license       = { :type => 'MIT', :file => 'LICENSE' }

    s.authors = {
        'Maxim Khatskevich' => 'maxim@khatskevi.ch'
    } # authors

    s.swift_version = '4.2'

    s.cocoapods_version = '>= 1.5.3'

    # === ios

    s.ios.deployment_target = '9.0'

    # === watchos

    s.watchos.deployment_target = '3.0' // THIS line causes problems!

    # === tvos

    s.tvos.deployment_target = '9.0'

    # === osx

    s.osx.deployment_target = '10.11'

    # === SUBSPECS ===

    s.subspec 'Core' do |ss|

        ss.source_files = 'Sources/Core/**/*.swift'

    end # subspec 'Core'

    s.subspec 'Operators' do |ss|

        ss.dependency 'XCEPipeline/Core'
        ss.source_files = 'Sources/Operators/**/*.swift'

    end # subspec 'Operators'

    s.test_spec 'Tests' do |ss|

        ss.requires_app_host = false
        ss.source_files = 'Tests/**/*.swift'
        ss.framework = 'XCTest'
        ss.dependency 'SwiftLint'

    end # test_spec 'Tests'

end # spec s
```

When I want to validate my spec, I run:

```bash
bundle exec pod lib lint
```

## What did you expect to happen?

The pod to pass validation.

## What happened instead?

Validation fails.

```bash
> XCEPipeline (1.1.2)
    - NOTE  | [XCEPipeline/Core,XCEPipeline/Operators] xcodebuild:  note: Using new build system
    - NOTE  | [XCEPipeline/Core,XCEPipeline/Operators] xcodebuild:  note: Planning build
    - NOTE  | [XCEPipeline/Core,XCEPipeline/Operators] xcodebuild:  note: Constructing build description
    - NOTE  | [XCEPipeline/Core,XCEPipeline/Operators] xcodebuild:  warning: Skipping code signing because the target does not have an Info.plist file. (in target 'App')
    - NOTE  | [watchOS] xcodebuild:  ld: warning: directory not found for option '-F/Applications/Xcode.app/Contents/Developer/Platforms/WatchSimulator.platform/Developer/Library/Frameworks'
    - ERROR | [watchOS] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - ERROR | [watchOS] xcodebuild:  /Users/maxim/Library/Mobile Documents/com~apple~CloudDocs/Dev/XCEssentials/Pipeline/Tests/PipelineTests.swift:27:8: error: no such module 'XCTest'

[!] XCEPipeline did not pass validation, due to 2 errors.
You can use the `--no-clean` option to inspect any issue.
```

### Stack

```
   CocoaPods : 1.6.0.beta.2
        Ruby : ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]
    RubyGems : 2.7.7
        Host : Mac OS X 10.13.6 (17G3025)
       Xcode : 10.1 (10B61)
         Git : git version 2.19.0
Ruby lib dir : /Users/maxim/.rvm/rubies/ruby-2.5.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 84791d40c2fa3b3cb95ed72bb6a39d3ec77308c1
```

### Installation Source

```
Executable Path: /Users/maxim/.rvm/gems/ruby-2.5.1/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-generate    : 1.3.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

## Project that demonstrates the issue

The spec listed above comes from this repo:
[XCEPipeline](https://github.com/XCEssentials/Pipeline)

NOTE: if we remove 'watchOS' support, then it passes validation:
[Working XCEPipeline.podspec](https://github.com/XCEssentials/Pipeline/blob/master/XCEPipeline.podspec)

**Comment by SlaunchaMan** [comment]
I was able to solve this by manually specifying the platform in my test spec: 

```Ruby
s.test_spec 'IRLSizeTests' do |ts|
    ts.platform = :ios, '10.0'
```

**Comment by maximkhatskevich** [comment]
@SlaunchaMan yes, I know this trick, but here are my thoughts about this.

1. We should not use workarounds, most logical interpretation of the spec must be built into the CocoaPods itself.
2. My lib supports all 4 Cocoa platforms, as it only depends on standard Swift lib. So the unit tests code is identical for all platforms (as well as lib sources). So using this approach means I need to create 3 test specs (excluding one for watchOS), which are going to be configured identical except deployment target, which is obviously inconvenient and error prone, hence should be avoided.
3. This workaround causes spec to pass validation, but with lots of annoying warnings from linter — for each supported spec platform it reports that incompatible test targets are ignored as incompatible. It unnecessary pollutes the output, which is always unwelcome.
4. I also use [cocoapods-generate](https://github.com/square/cocoapods-generate) plugin, which, unfortunately fails when this workaround is in use - for some reasons it attempts to create and build a unit tests target for watchOS and, of course, can not find 'XCTest' and fails the whole process (while when test spec is declared as in my podspec file example above it only creates unneded unit tests target for watchOS, but does not fail the whole process).

To sum it up, overall I think this behavior would be fare to call "unexpected" (at least), should be considered as undesired in CocoaPods and is worth to be fixed/improved.

Dear CocoaPods contributers, please, feel free to comment and let me know what is your point of view on this. Thanks!

**Comment by segiddins** [comment]
As a workaround in the meantime, you can specify every platform but watchOS in your test spec

**Comment by bitomule** [comment]
Setting an specific version inside the tests spec doesn´t seem to work for me whey I do a pod lib lint, still get the error.

```` - ERROR | [watchOS] unknown: Encountered an unknown error (The platform of the target `App` (watchOS 4.0) is not compatible with `APIClient/Tests (1.0)`, which does not support `watchos`.) during validation. ````

**Comment by maximkhatskevich** [comment]
Any updates on this issue, any idea when it's gonna be resolved? Thanks!

**Comment by dnkoutso** [comment]
keep. No updates here.

**Comment by paulb777** [comment]
The issue seems to be generating a main.m in the generated App target - similar to https://github.com/square/cocoapods-generate/issues/42

```
    Testing failed:
    	Entry point (_main) undefined. for architecture i386
    	Testing cancelled because the build failed.
```

**Comment by paulb777** [comment]
It seems that CocoaPods generates an empty main.m in the AppHost-*-UnitTests target in the watchOS target. If I work around that, I see that XCTest doesn't support watchOS.  Given that, it's probably a low priority to fix this for now.  

In the meantime, disable watchOS for the testspecs as noted [above](https://github.com/CocoaPods/CocoaPods/issues/8283#issuecomment-439503006).

![Screen Shot 2019-12-07 at 2 31 15 PM](https://user-images.githubusercontent.com/73870/70381431-dd31a700-18fe-11ea-92bf-e22031c1d29f.png)

**Comment by chliangGoogle** [comment]
The workaround also doesn't work for me. I set all the platform except for watch and I got the same error:
[!] The platform of the target `App-watchOS` (watchOS 6.0) is not compatible with `FirebaseMessaging/unit (4.1.9)`, which does not support `watchos`.

Here's how I set up the platform:
s.test_spec 'unit' do |ts|
    ts.platforms = {:ios => '8.0', :osx => '10.11', :tvos => '10.0'}

**Comment by paulb777** [comment]
Reopening now that Xcode 12.5 has added XCTest for watchos.

**Comment by JohnEstropia** [comment]
Issues still exists on iOS 16, Xcode 14. Tried all workarounds mentioned here with no success
`pod lib lint`
```
...
...
    - ERROR | [watchOS] xcodebuild: Returned an unsuccessful exit code.
    - NOTE  | [watchOS] xcodebuild:  xcodebuild: error: Unable to find a destination matching the provided destination specifier:
...
...
[!] Skipping test spec `CoreStore/CoreStoreTests` on platform `watchos` since it is not supported.
```
